server.js

```diff
@@ -1,77 +1,106 @@
 redisClient.on('connect', () => {
-    console.log('REDIS OK');
+  console.log('REDIS OK');
 });
 
 /*
  * REAL MAGIC HAPPENS
  */
+client.on('message', (message) => {
+  if (message.content.match(IS_URL_REGEX)) {
+    const hashedUrl = SHA256(message.content).toString();
 
-client.on('message', message => {
-    if (message.content.match(IS_URL_REGEX)) {
-        const hashedUrl = SHA256(message.content).toString();
-
-        redisClient.get(hashedUrl, (err, reply) => {
-            if (!reply) {
-                let data = {
-                    'user': message.author.username,
-                    'channel': message.channel.name,
-                    'date': message.createdAt
-                };
-                redisClient.set(hashedUrl, JSON.stringify(data), 'EX', 43200);
-            } else if (message.deletable) {
-                let data = JSON.parse(reply);
-                let date = moment(new Date(data.date));
-                let additionalMessage = data.user + ' posted this in #' + data.channel + ' on ' + date.fromNow() || '';
-                message.channel.send(additionalMessage, {file: __dirname+'/static/img/no_repost.jpg'});
-                message.delete();
-            }
+    redisClient.get(hashedUrl, (err, reply) => {
+      if (!reply) {
+        // link is fresh or at least 12 hours old
+        // save the hash and links meta in redis for 12 hours
+        const data = {
+          user: message.author.username,
+          channel: message.channel.name,
+          date: message.createdAt,
+        };
+        redisClient.set(hashedUrl, JSON.stringify(data), 'EX', TWELVE_HOURS);
+      } else if (message.deletable) {
+        // IT'S A REPOST :SCREAM:
+        // delete the message, post the repost PNG
+        // and blame the user with a custom blame message
+        const data = JSON.parse(reply);
+        const date = moment(new Date(data.date));
+        const blameMessage =
+          `${data.user} posted this in #${data.channel} on ${date.fromNow()}` || '';
+        message.channel.send(blameMessage, {
+          file: `${PATH_TO_STATIC_FOLDER}/no_repost.jpg`,
         });
-    }
+        message.delete();
+      }
+    });
+  }
 
-    if (message.content.match(IS_AH_REGEX)) {
-        message.channel.send('', {file: __dirname+'/static/img/ah.png'});
-    }
+  if (message.content.match(IS_AH_REGEX)) {
+    message.channel.send('', { file: `${PATH_TO_STATIC_FOLDER}/ah.png` });
+  }
 
-    if (message.content.replace(/\s/g,'').match(IS_NESTCEPAS_REGEX)) {
-        message.channel.send('', {file: __dirname+'/static/img/nestcepas.gif'});
-    }
+  if (message.content.replace(/\s/g, '').match(IS_NESTCEPAS_REGEX)) {
+    message.channel.send('', { file: `${PATH_TO_STATIC_FOLDER}/nestcepas.gif` });
+  }
+
+  if (message.content.replace(/\s/g, '').match(IS_PROJET_REGEX)) {
+    message.channel.send('', { file: `${PATH_TO_STATIC_FOLDER}/projet.gif` });
+  }
+
+  if (message.content.match(IS_GET_VANCOUVER_TIME_REGEX)) {
```

Comment:
I think we can make a function as it is repeated many times ?

Reply:
Yup totally, and we should add some test too. I want to add some CI server with unit testing etc. I know that's totally overkill but i think it's a good way to learn 😄